### DISCUSSION TITLE
SD Player and SPI recommendations

### DISCUSSION THREAD

**Discussion by y-nk** [pschatzmann/arduino-audio-tools#1715] [Q&A] [answered]
Hello,

As I understand, SPI bus can be shared although some selection need to be done to the device currently using the bus. I'd like to know what is the recommendations towards this, regarding the sd player. Is it mandatory for the player to have its own dedicated bus (for streaming) or can it be shared with others (like a mfrc522 for instance).

Thanks a lot for your answers

**Comment by pschatzmann** [comment, marked as answer]
It is recommended to use a dedicated SPI out of performance reasons.

You wiill need to check your individual case if you can get away by using a shared SPI: e.g 
- by using compressed data (mp3) 
- by avoiding 24/32 bit uncompressed data 
- by keeping the sample rate low
- by optimizing the SPI bit clock rate

**Reply by y-nk** [reply]
thanks a lot for the prompt answer @pschatzmann. i had the same impression that streaming would imply dedicated spi bus.
